### PR TITLE
website: TF-153: Break up expressions, resources, and modules pages

### PR DESCRIPTION
**Pull request by nfagerlund**
This is the next stage of reorganizing the language and CLI docs! The stage where I break a couple of links. 

This PR smashes apart the following huge pages into more navigable chunks:

- https://www.terraform.io/docs/configuration/resources.html
- https://www.terraform.io/docs/configuration/expressions.html
- https://www.terraform.io/docs/configuration/modules.html

(For the full effect, I recommend zooming your browser out as far as you can, so you can see more of each page on screen at once.) 

Since these pages have a lot of important content, they've accumulated a very large number of deep links to specific in-page anchors. Those kinds of anchor links cannot be redirected at the HTTP layer, because they're handled privately by the web browser. (It's feasible to redirect them using some JavaScript trickery, but I'm trying to not block this work on new website feature development.) I've updated all internal links within the core Terraform docs, but I expect a lot of external links exist.

So, instead of redirecting the exploded pages, I've converted them to landing pages that link to where every notable chunk of content has gotten off to. These landing pages preserve all of the original anchor IDs, so existing deep links should land directly at a link to what you were originally trying to learn. The landing pages are also "ghost" pages, which display the language sidebar nav but don't appear in it. 

In the case of the expressions page, I tried to write the landing page blurbs in such a way that we might later repurpose them for an "index of all syntax" page. 

## Testing

To browse a local build of this PR, you'll need to:

- Check out [this terraform-website branch](https://github.com/hashicorp/terraform-website/tree/oct20_support_language_and_cli_docs).
- Make sure the `ext/terraform` submodule is initted, then go into the submodule and `git fetch origin pull/26902/head && git checkout FETCH_HEAD`.
- Return to the top of terraform-website and run `make website`.

## tl;dr

Screen Shot 2020-11-12 at 6 51 13 PM Screen Shot 2020-11-12 at 6 51 03 PM Screen Shot 2020-11-12 at 6 50 47 PM

### CODE REVIEW

**Comment by nfagerlund** [comment]
An answer to a question that will probably occur to someone: Yes, I went through the commit delta from when I originally did this split-up work to the point where I rebased it, and I made sure to transfer every upstream change to these pages into the new locations. There weren't too many, but the bit about for_each's behavior with sets was mildly important.

**Comment by nfagerlund** [comment]
Also, I did some revisions of the text for each split-up section, but I didn't arrange the commits to make it easy to review those revisions. If you'd like me to rig something up for that, please let me know! I can probably reverse-engineer it. But I thought it was most important to get this up where it could be browsed in its "done" form.

**Comment by nfagerlund** [comment]
(Rebasing to fix conflicts -- an upstream PR removed a repeated chunk from the "references to named values" part of expressions, but I had already removed that in this PR.)

**Comment by alisdair** [comment]
> Make sure the `ext/`terraform submodule is initted, then go into the submodule and `git fetch origin pull/26723/head && git checkout FETCH_HEAD`.

I was really confused about why this didn't work for a bit. I think the PR number in this command should be 26902 instead.

**Comment by nfagerlund** [comment]
@alisdair WHOOPS, thank you! That was a copy-paste error from the previous PR. 😬

**Review by alisdair — APPROVED** [review]
This looks great to me! 👏 I love the unification of meta-argument definitions across resources & modules, and I really like the use of landing pages to preserve anchored permalinks.

**Review by apparentlymart — APPROVED** [review]
This looks good to me!

Given the size of this changeset and that it touches some files that we touch pretty often in other work, I recommend that we try to merge this ASAP and then tackle any feedback on the details in subsequent smaller PRs, just to reduce the risk of conflicts between this and other PRs. (It's still possible to leave inline comments on the diff even after the PR is merged.)

Once consequence of merging this is that it'll become harder to backport changes that include documentation updates into earlier maintenance branches, but that seems like one of those things where there'll never be a _great_ time to do it, and so I'd sooner just get on with it.

**Review by judithpatudith — APPROVED** [review]
I'm happy with the merge plan. 

I really like the solution of hidden pages to preserve old deep links, but the UX is a little crunchy. The landing pages are pretty short and as a result it's a bit hard to parse which section a user should look at when they come in because of a particular anchor. 

For example when you visit an anchor on the modules page you can see the whole page, so it isn't clear what section you should click on exactly.

It might be nice to make the anchors a little more transparent by adding a more detailed list of topics covered in each section that mentions each included anchor specifically. I hope that makes sense and am definitely open to other solutions.
